tests/test_study.py

```diff
@@ -528,3 +532,28 @@ def test_load_study(storage_mode, cache_mode):
         # Test loading an existing study.
         loaded_study = optuna.study.load_study(study_name=study_name, storage=storage)
         assert created_study.study_id == loaded_study.study_id
+
+
+@pytest.mark.parametrize('storage_mode', STORAGE_MODES)
+def test_running_study(storage_mode):
+    # type: (str) -> None
+
+    with StorageSupplier(storage_mode) as storage:
+        study = optuna.create_study(storage=storage)
+
+        # Run ten trials.
+        study.optimize(lambda t: t.suggest_int('x', 0, 10), n_trials=10)
+
+        # The methods of `InTrialStudy` behave the same as the counterparts at `Study`.
+        in_trial_study = InTrialStudy(study)
+        assert in_trial_study.best_params == study.best_params
+        assert in_trial_study.best_value == study.best_value
+        assert in_trial_study.best_trial == study.best_trial
+        assert in_trial_study.direction == study.direction
+        assert in_trial_study.trials == study.trials
+
+        assert in_trial_study.system_attrs == {}
+
+        in_trial_study.set_system_attr('foo', 'bar')
+        assert in_trial_study.system_attrs == {'foo': 'bar'}
+        assert in_trial_study.system_attrs == study.system_attrs
```

Comment:
Please add a test case for `in_trial_study.optimze()` to prevent careless mistakes in future development.

Reply:
I updated this unit test at the commit https://github.com/pfnet/optuna/pull/380/commits/0bdb96a65424388fe353351de1abc77bab8b7c2b.